### PR TITLE
All the methods were tested

### PR DESCRIPTION
**Pull request by jnicolaslaurenti**
Test completed (update):

image

Coverage:

Presenter:

image

Model (update):

image

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -31,12 +31,37 @@ android {
...
+    testImplementation 'junit:junit:4.13.1'
+
     androidTestImplementation 'androidx.test.ext:junit:1.1.2'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
+    androidTestImplementation('androidx.test.espresso:espresso-core:3.1.0', {
+        exclude group: 'com.android.support', module: 'support-annotations'
+    })
+
+
+    /*
```
Comment (LuchoElissondoGlobant) [inline app/build.gradle]: delete comented codee plis

Comment (jnicolaslaurenti) [reply app/build.gradle]: Solved!

File: app/src/main/java/com/onboarding/calculator/util/ConstantsUtils.java [hunk 2]
```diff
@@ -20,4 +20,15 @@
     public static final int CORRECTION_FACTOR = 1;
     public static final double DEFAULT_RESULT = 0.0;
 
+    //TEST
```
Comment (mauroLonghi) [inline app/src/main/java/com/onboarding/calculator/util/ConstantsUtils.java]: remove this comment line, because you have the sufix "TEST" in the constants

Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/util/ConstantsUtils.java]: Solved!

File: app/src/test/java/com/onboarding/calculator/CalculatorTest.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        verify(view).operationViewUpdate(model.getLastModified());
+        verifyNoMoreInteractions(view);
+    }
+
+    @Test
+    public void setValuesInFirstOperand() {
+        //Test
+        presenter.onCalculatorButtonPressed(ONE_STRING_TEST);
+        //Assert
+        assertEquals(ONE_STRING_TEST, model.getLastModified());
```
Comment (mauroLonghi) [inline app/src/test/java/com/onboarding/calculator/CalculatorTest.java]: here, you can add another control if it is saved into the firstOperand no?

File: app/src/test/java/com/onboarding/calculator/CalculatorTest.java [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+        assertEquals(TWO_STRING_TEST, model.getLastModified());
+        verify(view).showValues(model.getLastModified());
+        verify(view).operationViewUpdate(model.getOperation());
+        verifyNoMoreInteractions(view);
+    }
+
+    @Test
+    public void CleanAllTheOperation() {
+        presenter.deleteAll();
+        assertEquals(EMPTY_STRING, model.getOperation());
```
Comment (mauroLonghi) [inline app/src/test/java/com/onboarding/calculator/CalculatorTest.java:80]: the same at the previous commment, check not only the operation, intead check if all the field are empty

File: app/src/test/java/com/onboarding/calculator/CalculatorTest.java (hunk 3 above)
Comment (jnicolaslaurenti) [reply app/src/test/java/com/onboarding/calculator/CalculatorTest.java]: Solved in a call!

File: app/src/test/java/com/onboarding/calculator/CalculatorTest.java (hunk 4 above)
Comment (jnicolaslaurenti) [reply app/src/test/java/com/onboarding/calculator/CalculatorTest.java:80]: Solved in a call!

**Review by BQuinteros — APPROVED** [review]

File: app/src/test/java/com/onboarding/calculator/CalculatorTest.java [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+    }
+
+    @Test
+    public void operationWithoutErrors() {
+        model.setValues(ONE_STRING_TEST);
+        model.setOperator(MUL);
+        model.setValues(TWO_STRING_TEST);
+        Double result = model.getResult();
+        presenter.onEqualsButtonPressed();
+        assertEquals(Double.valueOf(2), result);
```
Comment (nicolbidone) [inline app/src/test/java/com/onboarding/calculator/CalculatorTest.java]: you missed a const here

Comment (jnicolaslaurenti) [reply app/src/test/java/com/onboarding/calculator/CalculatorTest.java]: Solved!

**Review by nicolbidone — APPROVED** [review]

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java [hunk 6]
```diff
@@ -89,60 +90,57 @@ private boolean operandEnabled(String operand) {
...
+            }
+        }
+        return result;
     }
 
     @Override
     public Double getResult() {
-        Double result = DEFAULT_RESULT;
-        if (!firstOperand.equals(EMPTY_STRING) && !firstOperand.equals(SUB)) {
+        Double result = null;
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: why do you use a null here? isn't better to use a 0.0 or something like that?

File: app/src/main/java/com/onboarding/calculator/util/ConstantsUtils.java [hunk 7]
```diff
@@ -20,4 +20,20 @@
     public static final int CORRECTION_FACTOR = 1;
     public static final double DEFAULT_RESULT = 0.0;
 
+    public static final String ZERO_STRING_TEST = "0";
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/calculator/util/ConstantsUtils.java]: Do not add constants for the tests here. Add them in a new constant utils class in the tests package or in the test class

**Review by SartiMau — CHANGES_REQUESTED** [review]

**Review by mauroLonghi — APPROVED** [review]

File: app/src/main/java/com/onboarding/calculator/util/ConstantsUtils.java (hunk 7 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/util/ConstantsUtils.java]: Solved!

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java (hunk 6 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: Solved!

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java [hunk 8]
```diff
@@ -89,60 +90,53 @@ private boolean operandEnabled(String operand) {
...
+
+            }
+            case MUL: {
+                return Double.parseDouble(firstOperand) * Double.parseDouble(secondOperand);
+            }
+            case DIV: {
+                return Double.parseDouble(firstOperand) / Double.parseDouble(secondOperand);
+            }
+        }
+        return null;
```
Comment (mauroLonghi) [inline app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: here you have the same problem returned a null...please use 0.0 or something

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java (hunk 8 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: Solved!

**Review by SartiMau — APPROVED** [review]

**Review by mauroLonghi — COMMENTED** [review]
waiting compile verification of the changes

**Review by mauroLonghi — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Pull request merged**